### PR TITLE
Export cross qk from beamsearch decoder subgraph.

### PR DESCRIPTION
**Pull request by zhanghuanrong**
Implement dtw() UnrollTensor() operators
Support extra decoding ids and no_speech_probs feature.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py [hunk 1]
```diff
@@ -149,7 +214,34 @@
...
+            inputs=["no_speech_probs_beam"],
+            outputs=["no_speech_probs"],
+            name="no_speech_probs_cast_to_fp32",
+            to=TensorProto.FLOAT,
+        )
+        graph_nodes.extend([prob_cast_node])
     beam_graph = helper.make_graph(graph_nodes, "beam-search-test", graph_inputs, graph_outputs, initializers)
+
+    if args.cross_qk_onnx_model:
+        print(f"*****Merging post model after BeamSearch*****")
```
[the remark this answers is not in the document]
Comment (zhanghuanrong) [reply onnxruntime/python/tools/transformers/models/whisper/whisper_chain.py]: removed

**Comment by tianleiwu** [comment]
Could you put UnfoldTensor and DynamicTimeWarping to separated PRs? Also add some tests for DynamicTimeWarping and this cross qk.

File: onnxruntime/core/graph/contrib_ops/bert_defs.cc [hunk 2]
```diff
@@ -613,6 +613,10 @@ ONNX_MS_OPERATOR_SET_SCHEMA(
               "The value to be filled in the attention mask. Default value is -10000.0f",
               AttributeProto::FLOAT,
               OPTIONAL_VALUE)
+        .Attr("output_qk",
```
Comment (tianleiwu) [inline onnxruntime/core/graph/contrib_ops/bert_defs.cc:616]: Do not need this attribute. You can deduce the flag from whether there is 3rd output

File: onnxruntime/core/graph/contrib_ops/contrib_defs.cc [hunk 3]
```diff
@@ -1081,6 +1124,10 @@ ONNX_MS_OPERATOR_SET_SCHEMA(BeamSearch, 1,
                                       "Size of the vocabulary. "
                                       "If not provided, it will be inferred from the decoder subgraph's output shape",
                                       AttributeProto::INT, static_cast<int64_t>(-1))
+                                .Attr("decoder_output_cross_qk", "If nozero, decoder subgraph contains output Q*K from cross attentions. Default 0.", AttributeProto::INT, OPTIONAL_VALUE)
```
Comment (tianleiwu) [inline onnxruntime/core/graph/contrib_ops/contrib_defs.cc:1127]: Since it need so many changes in BeamSearch interface, I suggest to add a new operator.

Comment (zhanghuanrong) [reply onnxruntime/core/graph/contrib_ops/contrib_defs.cc:1127]: Yeah. worth to make a separate op for this.

**Pull request closed without merging**